blueflood-core/src/main/java/com/rackspacecloud/blueflood/rollup/Granularity.java

```diff
@@ -254,23 +280,25 @@ private static Granularity granularityFromPointsLinear(double requestedDuration,
      *
      * @param requestedDuration (milliseconds)
      */
-    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints) {
+    private static Granularity granularityFromPointsGeometric(String tenantid, long from, long to, double requestedDuration, int requestedPoints, long assumedIntervalMillis, boolean checkTtl) {
         double minimumPositivePointRatio = Double.MAX_VALUE;
         Granularity gran = null;
-        if (SAFETY_TTL_PROVIDER == null) {
+        if (checkTtl && SAFETY_TTL_PROVIDER == null) {
             SAFETY_TTL_PROVIDER = SafetyTtlProvider.getInstance();
         }
 
```

Comment:
@ChandraAddala This is intended to be related to the testing. Please, see above. The overload with the `checkTtl` parameter is only called by the tests. The changes only minimally change the functionality, and only when the tests are running. Would it be better if I added a `@VisibleForTesting` annotation? Failing that, I could create a separate story to add the overload.